store/tikv/2pc.go

```diff
@@ -273,9 +264,10 @@ func (c *twoPhaseCommitter) doActionOnBatches(bo *Backoffer, action twoPhaseComm
 
 	// Concurrently do the work for each batch.
 	ch := make(chan error, len(batches))
-	for _, batch := range batches {
-		go func(batch batchKeys) {
-			reserveStack(false)
+	for _, batch1 := range batches {
+
+		batch := batch1
```

Comment:
why not just use batch1?

Reply:
That's a good habit ! @winkyao
When using closure along with range operation, it's easy to get suck:

```
	arr := []int{1, 2, 3, 4}
	for _, v := range arr {
		go func() {
			fmt.Println(v)   // guess the result?
		}()
	}
```

Reply:
got it!